### ISSUE TITLE
老师您好，Model的evaluate接口，batch_size的大小不同，输出的acc也不同，请问是什么原因

### ISSUE THREAD

**Issue by Renxg150** [PaddlePaddle/Paddle#40095]
aistudio地址：https://aistudio.baidu.com/aistudio/projectdetail/3524260?contributionType=1
代码片段1：

`# 用 evaluate 在测试集上对模型进行验证

eval_result = model.evaluate(test_dataset,batch_size=391, verbose=1)`

输出：

`Eval begin...

step 1/1 [==============================] - loss: 3.0728 - acc: 0.5857 - 14ms/step

Eval samples: 391`

代码片段2：

`# 用 evaluate 在测试集上对模型进行验证

eval_result = model.evaluate(test_dataset, verbose=1) # 默认batchsizee为1`

输出：

`Eval begin...

step 391/391 [==============================] - loss: 3.6563 - acc: 0.0486 - 6ms/step  
        
Eval samples: 391`

这两种方式进行验证的acc差距十分的大，请问我应该相信哪种结果，以及相差如此大的差距的原因。万分期待您的解答，老师。

**Comment by gongel** [comment]
官方给个例子没有问题哈，你可以运行看看，然后check一下你自己的程序。https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/Model_cn.html#evaluate-eval-data-batch-size-1-log-freq-10-verbose-2-num-workers-0-callbacks-none

**Comment by GT-ZhangAcer** [comment]
咱们高层API在输出评估指标的时候，是输出瞬时指标还是平均指标呢？感觉换成平均指标会好一点，瞬时的偶然程度有点大，很容易影响到开发者做判断~

**Comment by Renxg150** [comment]
老师您好，我学习了您提供的api文档，想问一下model.evaluate中的batch_size的值和输出的正确率以及loss有什么联系，万分感谢。

**Comment by Renxg150** [comment]
> 官方给个例子没有问题哈，你可以运行看看，然后check一下你自己的程序。https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/Model_cn.html#evaluate-eval-data-batch-size-1-log-freq-10-verbose-2-num-workers-0-callbacks-none

老师您好，我学习了您提供的api文档，想问一下model.evaluate中的batch_size的值和输出的正确率以及loss有什么联系，万分感谢。

**Comment by Renxg150** [comment]
> 咱们高层API在输出评估指标的时候，是输出瞬时指标还是平均指标呢？感觉换成平均指标会好一点，瞬时的偶然程度有点大，很容易影响到开发者做判断~

老师您好，感谢您的回复，请问您说的瞬时指标和平均指标是什么含义，十分期待您的点拨，万分感谢。

**Comment by Renxg150** [comment]
飞桨的老师们好，给老师们反馈一下前几天我遇到的问题，正确率随着batch_size的大小改变那个问题现在已经解决。原因是我在模型里使用了paddle.static.nn.batch_norm，应该使用paddle.nn.BatchNorm，这样就会锁定这个层的均值和方差了。感谢老师们的帮助。
